### PR TITLE
Added the XKCD antigravity comic in ASCII form (2)

### PR DESCRIPTION
**Pull request by JoeGlancy**
I re-created the XKCD antigravity comic (http://xkcd.com/353/) in ASCII form, and added it to the antigravity module to simulate the classic Python antigravity. There's two versions, the micro one (default), and the full one (optionally swappable for the micro one at build time). It uses the RLE as suggested by @whaleygeek and @dpgeorge to encode the strings. The micro comic, both text and RLE printing code, is 288 bytes. The larger comic, both text and RLE code, is just over 1KB (1053 bytes).

### CODE REVIEW

**Comment by dpgeorge** [comment]
@JoeGlancy really great!  Just a few code style issues to fix up, if you don't mind: 4 spaces for tab, curly brackets on the line with the statement, and curly brackets even for 1-liner.  Also please squash all commits for the final merge (and rebase against master: git rebase -i master).

File: source/microbit/modantigravity.cpp [hunk 1]
```diff
@@ -4,7 +4,7 @@
  *
  * The MIT License (MIT)
  *
- * Copyright (c) 2015 Damien P. George
+ * Copyright (c) 2015-2016 Damien P. George, Joe Glancy
```
Comment (dpgeorge) [inline source/microbit/modantigravity.cpp]: Please put different names on different lines, ie add the following for you:
```
Copyright (c) 2016 Joe Glancy
```

File: source/microbit/modantigravity.cpp [hunk 2]
```diff
@@ -37,8 +37,140 @@ extern "C" {
 #define GET_PIXEL(x, y) microbit_display_get_pixel(&microbit_display_obj, x, y)
 #define SET_PIXEL(x, y, v) microbit_display_set_pixel(&microbit_display_obj, x, y, v)
 
+STATIC void antigravity_output_char(char c) {
+	MP_PLAT_PRINT_STRN((char*) &c, 1);
+}
+
+// NOTE: string has to be plain ASCII
+static void antigravity_print_rle(const char *s)
+{
```
Comment (dpgeorge) [inline source/microbit/modantigravity.cpp]: Put curly brace on same line as func name.

File: source/microbit/modantigravity.cpp [hunk 3]
```diff
@@ -37,8 +37,140 @@ extern "C" {
...
+	// RLE encoding format (2 characters, [0] (first) and [1] (second)):
```
Comment (dpgeorge) [inline source/microbit/modantigravity.cpp]: 4-space tabs.

File: source/microbit/modantigravity.cpp [hunk 4]
```diff
@@ -37,8 +37,140 @@ extern "C" {
...
+	{
+		if ((*s & 0x80) == 0x80)
+		{
+			// seventh bit set
+			reps = *s ^ 0x80;
+			s++;
+			if (*s == '\0')
+				return;
+			while (reps--)
+				antigravity_output_char(*s);
```
Comment (dpgeorge) [inline source/microbit/modantigravity.cpp]: Use curly braces for single-line block statments.

**Comment by JoeGlancy** [comment]
I think that's all.

**Comment by ntoll** [comment]
@JoeGlancy you need to add yourself to the AUTHORS file. ;-)

**Comment by dpgeorge** [comment]
Yes, looks good, but please combine commits into one.

**Comment by ntoll** [comment]
@JoeGlancy ``rebase -i master`` see https://help.github.com/articles/about-git-rebase/ :-)

**Comment by JoeGlancy** [comment]
I was sure I already did. I'll do both tonight when I find time.

**Comment by ntoll** [comment]
Great stuff and thanks! Looks like this'll make it into the imminent 1.0 release. :smile:

**Comment by JoeGlancy** [comment]
Sweet! I look forward to it :)

**Comment by ntoll** [comment]
Also, make sure you add yourself to the AUTHORS file before you roll up all the commits..!

**Comment by JoeGlancy** [comment]
Will do!

**Comment by JoeGlancy** [comment]
Done?

**Comment by dpgeorge** [comment]
Ok, great, it's now 256 bytes (the small version of the comic with 1 pane).

There are still 6 commits that need squashing, but I can do this with a rebase myself, unless there are objections?

**Comment by JoeGlancy** [comment]
Finally!
No, sorry, I attempted 3 times but it finally rebased successfully :)

**Comment by dpgeorge** [comment]
Great work, thanks for your contribution!  And hopefully you learned a bit more about using git :)

**Comment by JoeGlancy** [comment]
Yup :+1: :)

**Pull request merged**